### PR TITLE
Reset color and icon when disabled - issue #398

### PR DESCRIPTION
**Pull request by bakulf**

### CODE REVIEW

**Comment by groovecoder** [comment]
From the Travis job:
```
/home/travis/build/mozilla/testpilot-containers/index.js

  1487:58  error  Expected '!==' and instead saw '!='  eqeqeq

  1488:60  error  Expected '!==' and instead saw '!='  eqeqeq

  1493:76  error  Missing semicolon                    semi

  1499:78  error  Missing semicolon                    semi
```

**Comment by jonathanKingston** [comment]
The question to me is why are we storing preInstalledIdentities then breaking them? Is something not working here? I'm not sure we need more code to make this work do we?

**Comment by bakulf** [comment]
preInstalledIdentities contains just IDs. We cannot change it for compatibility reasons.

File: index.js [hunk 1]
```diff
@@ -1587,6 +1597,32 @@ ContainerWindow.prototype = {
 
     return true;
   },
+
+  _shutdownContainers() {
+    ContextualIdentityProxy.getIdentities().forEach(identity => {
```
Comment (jonathanKingston) [inline index.js]: Can we add a comment explaining this whole function.

`// Changes icons back to defaults when they are not in native firefox`

or something similar

File: index.js [hunk 2]
```diff
@@ -1587,6 +1597,32 @@ ContainerWindow.prototype = {
...
+      if (IDENTITY_ICONS_STANDARD.indexOf(identity.icon) !== -1 &&
+          IDENTITY_COLORS_STANDARD.indexOf(identity.color) !== -1) {
+        return;
+      }
+
+      if (identity.userContextId <= IDENTITY_ICONS_STANDARD.length) {
+        identity.icon = IDENTITY_ICONS_STANDARD[identity.userContextId - 1];
+      } else {
+        identity.icon = IDENTITY_ICONS_STANDARD[0];
```
Comment (jonathanKingston) [inline index.js]: We could actually check if the current icon doesn't exist in `IDENTITY_ICONS_STANDARD` which would keep icons how the user modified them.

Same for colors too.

**Comment by jonathanKingston** [comment]
> preInstalledIdentities contains just IDs. We cannot change it for compatibility reasons.

Bum! It's something we will have to remove in the SDK removal, is there any way it can be put in `background.js` instead? Even if we have to message that?

**Comment by jonathanKingston** [comment]
Rebased on the command line and merged in: https://github.com/mozilla/testpilot-containers/commit/3b9da05e673a531af964be4f332769d64ba0a134

**Pull request closed without merging**